ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,154 +83,260 @@ def test_h_max_results():
+        geom_TelescopeFrame = geom_CameraFrame.transform_to(
+            TelescopeFrame(telescope_pointing=telescope_pointings[tel_id])
+        )
 
-    assert reconstructed_events > 0
+        mask = tailcuts_clean(
+            geom_TelescopeFrame,
+            dl1.image,
+            picture_thresh=5.0,
+            boundary_thresh=2.5,
+            keep_isolated_pixels=False,
+            min_number_picture_neighbors=2,
+        )
 
+        try:
+            hillas_dict[tel_id] = hillas_parameters(
+                geom_TelescopeFrame[mask], dl1.image[mask]
+            )
 
-def test_invalid_events():
-    """
-    The HillasReconstructor is supposed to fail
-    in these cases:
-    - less than two teleskopes
-    - any width is NaN
-    - any width is 0
+            # the original event is created from a
+            # pytest fixture with "session" scope, so it's always the same
+            # and if we used the same event we would overwrite the image
+            # parameters for the next tests, thus causing their failure
+            test_event = deepcopy(event)
+            test_event.dl1.tel[tel_id].parameters = ImageParametersContainer()
+            test_event.dl1.tel[tel_id].parameters.hillas = hillas_dict[tel_id]
+
+        except HillasParameterizationError as e:
+            print(e)
+            continue
 
-    This test uses the same sample simtel file as 
-    test_reconstruction(). As there are no invalid events in this
-    file, multiple hillas_dicts are constructed to make sure 
-    Exceptions get thrown in the mentioned edge cases.
+    # Get shower geometry
+    reconstructor(event)
+    # get the result from the correct DL2 container
+    result = event.dl2.stereo.geometry["HillasReconstructor"]
 
-    Test will fail if no Exception or another Exception gets thrown."""
+    # get the reconstructed coordinates in the sky
+    reco_coord = SkyCoord(alt=result.alt, az=result.az, frame=AltAz())
+    # get the simulated coordinates in the sky
+    true_coord = SkyCoord(
+        alt=event.simulation.shower.alt, az=event.simulation.shower.az, frame=AltAz()
+    )
 
-    filename = get_dataset_path("gamma_test_large.simtel.gz")
+    # check that we are not more far than 0.1 degrees
+    assert reco_coord.separation(true_coord) < 0.1 * u.deg
 
-    fit = HillasReconstructor()
 
-    tel_azimuth = {}
-    tel_altitude = {}
+@pytest.mark.parametrize("filename", 
+                         ["gamma_divergent_LaPalma_baseline_20Zd_180Az_prod3_test.simtel.gz",
+                         "gamma_LaPalma_baseline_20Zd_180Az_prod3b_test.simtel.gz"])
+def test_CameraFrame_against_TelescopeFrame(filename):
 
-    source = EventSource(filename, max_events=10)
-    subarray = source.subarray
-    calib = CameraCalibrator(subarray)
+    input_file = get_dataset_path(
+        "gamma_divergent_LaPalma_baseline_20Zd_180Az_prod3_test.simtel.gz"
+    )
+
+    source = SimTelEventSource(input_file, max_events=10)
```

Comment:
Why does this need to be a `SimTelEventSource` specifically? Shouldn''t `EventSource` resolve to the correct one?

Reply:
If you know that you are looking at a simtel file, why rely on something guessing?

Reply:
Sure, was just wondering about the change

Reply:
Actually @LukasNickel has a point: recently @kosack told me that we should never use the low-level sources and always use the more general `EventSource`.

If this should apply everywhere, than it makes sense to do it also here?

Reply:
I would disagree with that. If you know what kind of file you are dealing with, there is no reason not to use the corresponding event source directly, I'd say. Explicit is better than implicit.

General tools expected to deal with arbitrary input are something else of course.